### DISCUSSION TITLE
Vendor specific host implementation

### DISCUSSION THREAD

**Discussion by maaamcube** [hathach/tinyusb#2496] [Q&A]
Hello,

I have three interconnected hubs and on each hub I have connected an STM32 whose USB is in vendor-specific device mode. I want to plug it to a pico w to establish wifi communication through usb.

Is there an example on vendor specific host implementation ?

Thanks.

**Comment by maaamcube** [comment]
I really need this stuff, can you help me how can i search to build by myself at least ?

**Reply by Mohanbandaru16** [reply]
did you find answers?

**Reply by maaamcube** [reply]
> did you find answers?

Nope.

**Reply by Mohanbandaru16** [reply]
i can read those usb's but could not able to communicate through end points

**Reply by maaamcube** [reply]
> i can read those usb's but could not able to communicate through end points

It is enumerated through vid and pid or kind of ? I think the communcation with ep is the complicated part for me also cause i don't have the documentation of that USB.

**Reply by Mohanbandaru16** [reply]
it shows all the vid pid and all the interfaces and end points but when im trying to communicate through end points i need to provide class drives like which one to use there im stuck .i need to write my own driver i think.

**Reply by maaamcube** [reply]
Yes, it is enumerated. I think also you have to write your own vendor if you have already the doc, it can help you much more, they not gonna write every vendor specific for every USB.

**Reply by Mohanbandaru16** [reply]
i have done this in esp32 s3  using class driver example and talk through end points but here in pico w i have no idea what to do.[https://github.com/espressif/esp-idf/tree/master/examples/peripherals/usb/host/usb_host_lib](url)

**Reply by maaamcube** [reply]
I tried this one also, it's enumerate the USB's with limitation(cause of the number of ep supported by the s3) but i can't talk to it.
Do you have a discord to talk their ?
